docs/internet-identity-spec.adoc

```diff
@@ -197,6 +197,8 @@ type DeviceData = record {
   pubkey : DeviceKey;
   alias : text;
   credential_id : opt CredentialId;
+  purpose : { #recovery; #authentication };
+  type : { #unknown; #platform; #crossPlatform; #seedPhrase };
```

Comment:
These should be wrapped in `opt`, no?

Reply:
We don’t have to be backward-compatible towards the frontend, so here we don’t.

We need to wrap them in `opt` when we serialize/deserialize to stable memory, because there we want backward compatibility.